src/module/actor/actions/base.ts

```diff
@@ -145,41 +145,47 @@ abstract class BaseAction<TData extends BaseActionVariantData, TAction extends B
         return new Collection(variants);
     }
 
-    protected async getDefaultVariant(options?: { variant?: string }): Promise<TAction> {
+    protected getDefaultVariant(options?: { variant?: string }): TAction {
         const variants = this.variants;
         if (options?.variant && !variants.size) {
             const reason = game.i18n.format("PF2E.ActionsWarning.Variants.None", {
                 action: game.i18n.localize(this.name),
                 variant: options.variant,
             });
-            return Promise.reject(reason);
+            throw new Error(reason);
```

Comment:
use `ErrorPF2e`

Reply:
I'm wondering if it makes more sense to just throw the string in this particular case. It's always caught bye the promise in the other methods calling it and is really meant to eventually go in a UI notification instead of the logs.

Reply:
for that we sometimes do `ui.notifications.error(ErrorPF2e("abcdefg").message)`;

Reply:
The Error never makes it out of the class - it is turned into a rejected promise. With the error I have to do an instanceof check to satisfy TS before extracting the message.